include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -304,4 +304,11 @@
 #    define _ONEDPL_ICPX_OMP_SIMD_DESTROY_WINDOWS_BROKEN 0
 #endif
 
+// Avoid compile errors due incorrect implementation of std::bit_floor in MS STL: used "extern" variable __isa_available
+#if defined(_MSC_VER) && _ONEDPL_BACKEND_SYCL
+#    define _ONEDPL_STD_BIT_FLOOR_BROKEN 1
```

Comment:
The MS implementation is not incorrect, it just cannot be used in SYCL kernels. Therefore some changes are needed: 
1. Change the condition to `(__SYCL_DEVICE_ONLY__ || __SYCL_SINGLE_SOURCE__)` in place of `_ONEDPL_BACKEND_SYCL`
2. Update the comment to e.g. "The implementation of std::bit_floor in MS STL does not meet requirements for SYCL device functions"

Reply:
Fixed

Reply:
Perhaps you need to still push?

Reply:
Forgot, pushed just now.